### PR TITLE
[Enhancement] Use concurrency map instead of lock for blacklist

### PR DESCRIPTION
**Pull request by ZiheLiu**
Why I'm doing:
The `lock` in `SimpleScheduler` only protects `BLACKLIST_BACKENDS`. The lock contention of `isInBlacklist` is pretty high for the high-concurrency scenario.

What I'm doing:
Therefore, use a concurrency map instead to avoid the lock contention of `isInBlacklist`.

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/qe/SimpleScheduler.java [hunk 1]
```diff
@@ -47,38 +47,38 @@
...
     //count id for backend get TNetworkAddress
-    private static AtomicLong nextBackendHostId = new AtomicLong(0);
+    private static final AtomicLong NEXT_BACKEND_HOST_ID = new AtomicLong(0);
+
     //count id for get ComputeNode
-    private static Map<Long, Integer> blacklistBackends = Maps.newHashMap();
-    private static Lock lock = new ReentrantLock();
-    private static UpdateBlacklistThread updateBlacklistThread;
-    private static AtomicBoolean enableUpdateBlacklistThread;
+    private static final ConcurrentMap<Long, Integer> BLACKLIST_BACKENDS = Maps.newConcurrentMap();
```
Comment (kevincai) [inline fe/fe-core/src/main/java/com/starrocks/qe/SimpleScheduler.java:73]: can change to `BLOCKLIST` together with this refactor.

Comment (ZiheLiu) [reply fe/fe-core/src/main/java/com/starrocks/qe/SimpleScheduler.java:73]: Do you mean rename `BLACKLIST_BACKENDS` to `BLACKLIST`?

Comment (kevincai) [reply fe/fe-core/src/main/java/com/starrocks/qe/SimpleScheduler.java:73]: literally repace `blacklist` with `blocklist`

Comment (ZiheLiu) [reply fe/fe-core/src/main/java/com/starrocks/qe/SimpleScheduler.java:73]: Got it.

Comment (ZiheLiu) [reply fe/fe-core/src/main/java/com/starrocks/qe/SimpleScheduler.java:73]: What's difference between blacklist and blocklist?

**Review by satanson — APPROVED** [review]

**Review by silverbullet233 — APPROVED** [review]

**Pull request merged**